python/ray/serve/gradio.py

```diff
@@ -0,0 +1,14 @@
+import gradio as gr
+import starlette
+
+from ray.serve.http_util import ASGIHTTPSender
+
+
+class GradioIngress:
+    def __init__(self, io: gr.Blocks):
+        self.app = gr.routes.App.create_app(io)
```

Comment:
Is this a public API that's guaranteed by gradio to remain stable?

Reply:
Yes, definitely. We need to make sure any APIs we use are public & stable otherwise our integration could break at any point without warning.

Reply:
Abubakar confirmed that this API will remain stable, and they will add tests around `gr.routes.App.create_app()` to make sure the API signature doesn't change!